labs-ios-starter/Application/Extensions/UIImageView+ImageDownloader.swift

```diff
@@ -6,17 +6,24 @@ import UIKit
 extension UIImageView {
     func downloaded(from url: URL, contentMode mode: UIView.ContentMode = .scaleAspectFill) {
         contentMode = mode
-        URLSession.shared.dataTask(with: url) { data, response, error in
+        
+        let networkService = NetworkService()
+        guard let request = networkService.createRequest(url: url, method: .get) else {
+            print("Failed to create request, check URL")
+            return
+        }
+        
+        networkService.dataLoader.loadData(using: request) { data, response, error in
             guard
-                let httpURLResponse = response as? HTTPURLResponse, httpURLResponse.statusCode == 200,
+                let httpURLResponse = response, httpURLResponse.statusCode == 200,
```

Comment:
NetworkService has changed quite a bit in the way loadData is called (using the response type). Also, the response code is already being checked before data (result.success) or error: ErrorHandler.NetworkError (result.failure) are returned to the caller in completion